src/components/pages/ImageClassification/ImageUploadModal.js

```diff
@@ -1,24 +1,124 @@
-import React, { useState, useRef } from 'react'
+import React, { useRef } from 'react'
 import PropTypes from 'prop-types'
 import Modal from '../../generic/Modal'
 import { ButtonPrimary } from '../../generic/buttons'
 import { DropZone, HiddenInput } from './ImageUploadModal.styles'
+import { toast } from 'react-toastify'
 
-const ImageUploadModal = ({ isOpen, onClose, onFilesUpload }) => {
-  const [selectedFiles, setSelectedFiles] = useState([])
+const ImageUploadModal = ({ isOpen, onClose, onFilesUpload, existingFiles }) => {
   const fileInputRef = useRef(null)
 
+  const validFileTypes = ['image/jpeg', 'image/pjpeg', 'image/png', 'image/mpo']
+  const maxFileSize = 30 * 1024 * 1024 // 30 MB
+  const maxWidth = 8000
+  const maxHeight = 8000
+
+  const validateDimensions = (file) => {
+    return new Promise((resolve) => {
+      const reader = new FileReader()
+      reader.onload = (event) => {
+        const img = new Image()
+        img.onload = () => {
+          if (img.width <= maxWidth && img.height <= maxHeight) {
+            resolve({ file, valid: true })
+          } else {
+            resolve({ file, valid: false })
+          }
+        }
+        img.onerror = () => {
+          resolve({ file, valid: false, corrupt: true })
+        }
+        img.src = event.target.result
+      }
+      reader.onerror = () => {
+        resolve({ file, valid: false, corrupt: true })
+      }
+      reader.readAsDataURL(file)
+    })
```

Comment:
@parmvirthind  are we uploading new files in the image annotation modal? in that case  it might be a good idea to add it in. it reads files asynchronously - can scan corrupt files. Al was testing different files before I added this in and certain file errors wouldn't be caught

Reply:
ok thought so - yup good without it then since we're only adding images through the image upload modal